contracts/apps/AragonApp.sol

```diff
@@ -57,6 +64,36 @@ contract AragonApp is AppStorage, Autopetrified, VaultRecoverable, ReentrancyGua
         );
     }
 
+    /**
+    * @dev Check whether a call to the current app can be executed or not based on the kill-switch settings
+    * @return Boolean indicating whether the call could be executed or not
+    */
+    function canExecuteCall() public view returns (bool) {
+        IKernel _kernel = kernel();
+        bytes4 selector = _kernel.shouldDenyCallingContract.selector;
+        bytes memory callData = abi.encodeWithSelector(selector, appId(), address(this));
+        bool success = address(_kernel).call(callData);
+
+        // if the call to `kernel.shouldDenyCallingApp` reverts (using an old Kernel) we consider that
+        // there is no kill switch and the call can be executed be allowed to continue
+        if (!success) return true;
+
+        // if not, first ensure the returned value is 32 bytes length
+        uint256 _outputLength;
+        assembly { _outputLength := returndatasize }
+        require(_outputLength == 32, ERROR_UNEXPECTED_KERNEL_RESPONSE);
```

Comment:
At least for now, we are using this check within a `require` statement, i.e. to raise or not an error. Thus, returning false or reverting here will result in the same effect. And I think that between having more context to know why a call reverted and not, I prefer having it. Do you have in mind some particular scenario where it could be useful swallowing that scenario?

Reply:
If someone were to call this function directly, outside of the modifier (e.g. the app itself or another external contract calling this function to see if it was killswitched or not).

One example is with the Token Manager's [`canForward()`](https://github.com/aragon/aragon-apps/blob/master/apps/token-manager/contracts/TokenManager.sol#L269), where if we don't bake-in the killswitch check at the `EVMScriptRunner` level, we'd have to do a query to `canExecuteCall()` there.

I'd also like to make `canPerform()` and `canExecuteCall()` as consistent as possible, so users interact with them using the same error handling model. `canPerform()` masks incorrect oracle param responses with `false`, rather than reverting.